uaclient/tests/test_cli_status.py

```diff
@@ -818,6 +853,122 @@ def test_unicode_dash_replacement_when_unprintable(
+        self,
+        _m_getuid,
+        m_get_contract_information,
+        _m_get_avail_resources,
+        _m_should_reboot,
+        _m_remove_notice,
+        exception_to_throw,
+        exception_type,
+        exception_message,
+        capsys,
+        FakeConfig,
+    ):
+        """Check that simulated status json/yaml output raises errors."""
+
+        m_get_contract_information.side_effect = exception_to_throw
+
+        cfg = FakeConfig()
+
+        args = mock.MagicMock(
+            format="json", all=False, simulate_with_token="some_token"
+        )
+
+        with pytest.raises(Exception) as exc:
+            action_status(args, cfg=cfg)
+
+        assert exc.type == exception_type
+        assert exception_message in exc.value.args
+
+    @pytest.mark.parametrize(
+        "token_to_use, warning_message",
+        (
+            (
+                "expired_token",
+                'Contract "some_id" expired on December 31, 2019',
+            ),
+            (
+                "token_not_valid_yet",
+                'Contract "some_id" is not effective until December 31, 9999',
+            ),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "format_type,event_logger_mode",
+        (("json", EventLoggerMode.JSON), ("yaml", EventLoggerMode.YAML)),
+    )
+    def test_warnings_for_token_dates(
+        self,
+        _m_getuid,
+        m_get_contract_information,
+        _m_get_avail_resources,
+        _m_should_reboot,
+        _m_remove_notice,
+        format_type,
+        event_logger_mode,
+        token_to_use,
+        warning_message,
+        capsys,
+        FakeConfig,
+        event,
+    ):
+        """Check that simulated status json/yaml output has warnings.
+
+        There must be warnings for expired tokens and for tokens not valid yet.
+        """
+
+        def contract_info_side_effect(cfg, token):
+            response = copy.deepcopy(RESPONSE_CONTRACT_INFO)
+            if token == "expired_token":
+                response["contractInfo"][
+                    "effectiveTo"
+                ] = "2019-12-31T00:00:00Z"
+            if token == "token_not_valid_yet":
+                response["contractInfo"][
+                    "effectiveFrom"
+                ] = "9999-12-31T00:00:00Z"
+            return response
+
+        m_get_contract_information.side_effect = contract_info_side_effect
+
+        cfg = FakeConfig()
+
+        args = mock.MagicMock(
+            format=format_type, all=False, simulate_with_token=token_to_use
+        )
+
+        with mock.patch.object(
+            event, "_event_logger_mode", event_logger_mode
+        ), mock.patch.object(event, "_command", "status"):
+            assert 0 == action_status(args, cfg=cfg)
+
+        if format_type == "json":
+            output = json.loads(capsys.readouterr()[0])
+        else:
+            output = yaml.safe_load(capsys.readouterr()[0])
+
+        assert output["warnings"][0]["message"] == warning_message
```

Comment:
Can we test the whole output here ? Just to double check that the full json/yaml format is being returned correctly

Reply:
Well the output is covered by other (unit and integration) tests. This one targets this key specifically - the proper way would even be to make sure the event was created instead of checking the output - we might want to change it along with many other tests